ui/proposals_page.go

```diff
@@ -0,0 +1,614 @@
+package ui
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"strconv"
+	"strings"
+	"time"
+
+	"gioui.org/layout"
+	"gioui.org/op/clip"
+	"gioui.org/op/paint"
+	"gioui.org/unit"
+	"gioui.org/widget"
+	"gioui.org/widget/material"
+
+	"github.com/ararog/timeago"
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/values"
+	"github.com/planetdecred/godcr/wallet"
+)
+
+const PageProposals = "Proposals"
+const (
+	categoryStateFetching = iota
+	categoryStateFetched
+	categoryStateError
+)
+
+type proposalNotificationListeners struct {
+	page *proposalsPage
+}
+
+func (p proposalNotificationListeners) OnNewProposal(proposal *dcrlibwallet.Proposal) {
+	p.page.addDiscoveredProposal(*proposal)
+}
+
+func (p proposalNotificationListeners) OnProposalVoteStarted(proposal *dcrlibwallet.Proposal) {
+	p.page.updateProposal(*proposal)
+}
+
+func (p proposalNotificationListeners) OnProposalVoteFinished(proposal *dcrlibwallet.Proposal) {
+	p.page.updateProposal(*proposal)
+}
+
+func (p proposalNotificationListeners) OnProposalsSynced() {
+	p.page.isSynced = true
+}
```

Comment:
This is what i intend to do in the next PR reason i requested we have this merged.
I have already started but it started getting complicated. Once this is merged it will be easier to migrate.